src/ingredients/ingredients.controller.ts

```diff
@@ -9,20 +9,30 @@ import {
   ParseIntPipe,
   Query,
 } from '@nestjs/common';
+
+import { Ingredient } from './entities/ingredient.entity';
 import { IngredientsService } from './ingredients.service';
 import { CreateIngredientDto } from './dto/create-ingredient.dto';
 import { UpdateIngredientDto } from './dto/update-ingredient.dto';
-import { PaginatedResponse } from '../common/interfaces/paginated-response';
-import { Ingredient } from './entities/ingredient.entity';
 import { PaginationQueryDto } from '../common/dto/pagination-query.dto';
+import { PaginatedResponse } from '../common/interfaces/paginated-response';
+import { Roles } from '../auth/decorators/roles.decorator';
+import { UserRole } from '../common/enums/user-role.enum';
+import { Auth } from '../auth/decorators/auth.decorator';
+import { GetUser } from '../common/decorators/get-user.decorator';
+import { User } from '../users/entities/user.entity';
 
 @Controller('ingredients')
 export class IngredientsController {
   constructor(private readonly ingredientsService: IngredientsService) {}
 
   @Post()
-  create(@Body() createIngredientDto: CreateIngredientDto) {
-    return this.ingredientsService.create(createIngredientDto);
+  @Auth(UserRole.Admin, UserRole.Moderator)
+  create(
+    @Body() createIngredientDto: CreateIngredientDto,
```

Comment:
Maybe it would be good to allow creating multiple ingredients too

Reply:
@tar-aldev sure, we should create separate task for this.